msteams-platform/m365-apps/build-actions-in-m365.md

```diff
@@ -0,0 +1,161 @@
+    this.setState({
+        itemId: itemId
+    });
+}
+```
+
+| &nbsp; | Name | Description |
+| --- | --- | --- |
+| &nbsp; | `actionObjects` | Array of corresponding action objects. |
+| &nbsp; | `itemId` | The app receives the ID as the content and uses it to query the Microsoft Graph. |
+
+## Access content through Graph API
+
+After obtaining the `itemId` of the triggering content, you can use the Graph API to read or modify the content, facilitating task completion for your users.
+
+```javascript
+app.getContext().then((context: app.Context) => {
+    const actionInfo = context.actionInfo;
+    if (actionInfo) {
+        // App was launched using an action    
+    } 
+    if (actionInfo && actionInfo.actionId == 'myActionId1') {
+        // Handle specific action    
+    } 
+    if (actionInfo) {
+        if (actionInfo.actionObject.type == app.ActionObjectType.M365Content) {
+            const itemId = actionInfo.actionObjects[0].itemId;
+            // Get the requested content from Mirosoft Graph by item id:
+        } 
+    }
+})
+```
+
+### Enable Actions in Microsoft Admin Center
+
+> [!NOTE]
+> Actions are available in [public developer preview](../resources/dev-preview/developer-preview-intro.md).
+
+To use an app with Actions in your tenant, an admin must upload the app package to the Microsoft Admin Center as follows:
+
+1. Go to [Microsoft Admin Center](https://admin.microsoft.com/AdminPortal#/homepage).
+
+1. Select **Settings** > **Integrated Apps** > **Upload custom apps**.
+
+Follow instructions to preinstall your app for entire organization or user groups within your tenant.
+
+## Sideload your app using Teams Toolkit
+
+> [!NOTE]
+> Actions are available in public developer preview, ensure that you enable [targeted release](/microsoft-365/admin/manage/release-options-in-office-365?view=o365-worldwide&preserve-view=true) to the users to experience Actions in Microsoft 365 app.
+
+After you update the app package with the required information, you're ready to test your Actions in the Teams Toolkit. To initiate debugging, select the **F5** key.
+
+   :::image type="content" source="images/actions-debug.png" alt-text="The screenshot shows actions in debug.":::
+
+A browser window opens with Microsoft 365 home page and your app is available under **Apps**.
+
+You can now preview your Actions in the Microsoft 365 home page, right-click a file that is supported by your Actions. Actions appear in the context menu, for example **Add todo task**.
+
+:::image type="content" source="images/actions-context-menu.png" alt-text="The screenshot shows the actions in context menu.":::
+
+## Code sample
+
+| **Sample name** | **Description** |**Node.js** |
+|-----------------|-----------------|----------------|
+| Actions in Microsoft 365 apps | This sample code describes the Actions in Microsoft 365 apps. | [View](https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main/samples/m365-actions-preview/nodejs)|
```

Comment:
Looking for better description:
This sample code contains two actions based on a To Do app.